xla/service/dump.cc

```diff
@@ -904,25 +894,43 @@ std::string GetNonDefaultDebugOptions(const DebugOptions& debug_options) {
   return non_default_options;
 }
 
-void DumpNonDefaultDebugOptions(const HloModule& module,
-                                absl::string_view suffix) {
+std::optional<std::string> DumpNonDefaultDebugOptions(
+    const HloModule& module, absl::string_view suffix) {
   const DebugOptions& debug_options = module.config().debug_options();
-  auto filename = FilenameFor(module, "", suffix);
-  auto nonDefaultDebugOptions = GetNonDefaultDebugOptions(debug_options);
-  DumpToFileInDir(debug_options, filename, nonDefaultDebugOptions);
+  std::string filename = FilenameFor(module, "", suffix);
+  std::string nonDefaultDebugOptions = GetNonDefaultDebugOptions(debug_options);
+  return DumpToFileInDirImpl(filename, nonDefaultDebugOptions,
+                             CanonicalDebugOptions(debug_options));
 }
 
-std::vector<std::string> DumpHloModuleIfEnabled(
-    const HloModule& module, const BufferAssignment& buffer_assn,
+std::vector<std::string> DumpHloModuleIfEnabledInternal(
```

Comment:
Put this function in the anonymous namespace. Also, replace "Internal" with "Impl" for consistency with other functions in the file.